// hooks/useFirebaseSet.js
import {useState, useEffect, useCallback} from 'react'
import firebase from 'firebase/app'
import 'firebase/firestore'
import useAuthChecker from './useAuthChecker'

export default () => {
  const [isLoading, setIsLoading] = useState(false)
  const [response, setResponse] = useState(null)
  const [error, setError] = useState(null)
  const [options, setOptions] = useState({})
  const [authUser] = useAuthChecker()

  const setDB = useCallback((options = {}) => {
    setOptions({
      ...options,
      authUser
    })
    setIsLoading(true)
  }, [])

  useEffect(() => {
    ;(async () => {
      const db = firebase.firestore()
      if (!isLoading || !authUser) return

      const post = {
        createdAt: Date.now(),
        description: "info info info info",
        id: 5,
        tagList: "tag1, tag2 ",
        title: "Post 5",
        username: "Artem",
        userCreate: ''
      }

      try {
        await db
          .collection(`posts/allPosts/`)
          .add(post)
        await db
          .collection(`posts/userPosts/${authUser.username}_${authUser.uid}`)
          .add(post)
        setResponse('success!')
      } catch (e) {
        setError(e)
      } finally {
        setIsLoading(false)
      }
    })()
  }, [isLoading, options, authUser])

  return [{isLoading, response, error}, setDB]

}
