// pages/Article.jsx
import React, {useEffect} from 'react'
import {Link} from 'react-router-dom'
import useFirebase from 'hooks/useFirebase'
import Loading from 'components/Loading'
import ErrorMessage from 'components/ErrorMessage'
import TagList from 'components/TagList'

const Article = ({match}) => {
  const postId = +match.params.slug
  const [{response, error, isLoading}, db] = useFirebase()

  useEffect(() => {
    db({
      url: '/posts',
      orderBy: 'createdAt',
      filter: 'tagsPost',
      userFilter: {
        fieldPath: 'id',
        opStr: '==',
        value: postId
      }
    })

  }, [db, postId])

  let post
  if (response) {
    post = response.posts[0][0]
  }

  return (
    <div className={'article-page'}>
      <div className="banner">
        {!isLoading && response && (
          <div className={'container'}>
            <h1>{post.title}</h1>
            <div className={'article-meta'}>
              <Link to={`/profiles/${post.username}`}>
                <img src={post.image} alt=""/>
              </Link>
              <div className="info">
                <Link to={`/profiles/${post.username}`}>
                  {post.username}
                </Link>
                <span className="date">{new Date(post.createdAt).toDateString()}</span>
              </div>
            </div>
          </div>
        )}
      </div>
      <div className="container page">
        {isLoading && <Loading/>}
        {error && <ErrorMessage/>}
        {!isLoading && response && (
          <div className={'row article-content'}>
            <div className="col-xs-12">
              <div>
                <p>{post.description}</p>
              </div>
              <TagList tags={post.tagList}/>
            </div>
          </div>
        )}
        <hr/>
      </div>
    </div>
  )
}

export default Article
