// pages/CreateArticle.jsx
import React, {useEffect, useState, useContext} from 'react'
import {Redirect} from 'react-router-dom'

import ArticleForm from 'components/ArticleForm'
import useFetch from 'hooks/useFetch'
import {CurrentUserContext} from 'contexts/currentUser'


const CreateArticle = () => {
  const apiUrl = '/articles'
  const [{response, error}, doFetch] = useFetch(apiUrl)
  const [currentUserState] = useContext(CurrentUserContext)
  const initialValue = {
    title: '',
    description: '',
    body: '',
    tagList: []
  }

  const [isSuccessFullSubmit, setIsSuccessFullSubmit] = useState(false)

  const handleSubmit = article => {
    doFetch({
      method: 'POST',
      data: {
        article
      }
    })
  }

  useEffect(() => {
    if (!response) {
      return
    }
    setIsSuccessFullSubmit(true)
  }, [response])

  if (!currentUserState.isLoggedIn) {
    return <Redirect to={'/'}/>
  }
  if (isSuccessFullSubmit) {
    return <Redirect to={`/articles/${response.article.slug}`}/>
  }

  return (
    <div>
      <ArticleForm
        errors={(error && error.errors) || {}}
        initialValue={initialValue}
        onSubmit={handleSubmit}
      />
    </div>
  )
}

export default CreateArticle
